pkg/reconciler/broker/broker.go

```diff
@@ -193,6 +173,29 @@ func (r *Reconciler) reconcileBroker(ctx context.Context, b *brokerv1beta1.Broke
 	}
 	b.Status.PropagateIngressAvailability(ingressEndpoints)
 
+	// Filter by `eventing.knative.dev/broker: <name>` here
+	// to get only the triggers for this broker. The trigger webhook will
+	// ensure that triggers are always labeled with their broker name.
+	triggers, err := r.triggerLister.Triggers(b.Namespace).List(labels.SelectorFromSet(map[string]string{eventing.BrokerLabelKey: b.Name}))
+	if err != nil {
+		logger.Error("Problem listing triggers", zap.Error(err))
+		return err
+	}
+
+	r.reconcileConfig(ctx, b, triggers)
+	// Update config map
+	r.flagTargetsForUpdate()
+	return nil
+}
+
+// reconcileConfig reconstructs the data entry for the given broker in targets-config.
+func (r *Reconciler) reconcileConfig(ctx context.Context, b *brokerv1beta1.Broker, triggers []*brokerv1beta1.Trigger) {
+	// TODO Maybe get rid of BrokerMutation and add Delete() and Upsert(broker) methods to TargetsConfig. Now we always
+	//  delete or update the entire broker entry and we don't need partial updates per trigger.
+	// The code can be simplified to r.targetsConfig.Upsert(brokerConfigEntry)
+	// First delete the broker entry.
```

Comment:
I had written a targets config reconciler which supports the interface you want here and allows directly constructing the broker config proto rather than working with the mutation interface: https://github.com/google/knative-gcp/pull/914. Directly constructing the proto is a bit nicer since we are not constrained to using the mutation interface and individual mutations do not need to be threadsafe. I had closed that since the brokercell reconciler was meant to replace this but maybe it is worth reviving after this PR is merged since the brokercell reconciler is not yet ready.

Reply:
Once we do 1054, we can start moving the target-config reconciliation to brokercell reconciler (assuming a static brokercell).

Reply:
I disagree that updating proto is better. Reconciler shouldn't worry about the internals of targets config. Again, I'm likely biased but just giving my voice.

Reply:
I am not sure I follow what do you mean by "internals" of targets config. Currently the `UpsertTargets` interface already takes in a `config.Target` object. What I am suggesting here is to add an interface to take in `config.Broker` because now we don't upsert each trigger one by one. If we think the `config.Broker` proto is "internal", we can always add another abstraction but I don't think it's needed.

Reply:
`config.Broker` is bit different from `config.Target` IMO. The "internal" I'm referring to is the map in the `config.Broker`. To construct a `config.Broker`, you will have to "manually" create the map rather than having the lib handle that. On the other side, `config.Target` is pure data.